aws/credentials.go

```diff
@@ -2,38 +2,25 @@ package aws
 
 import (
 	"context"
-	"math"
-	"sync/atomic"
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/internal/sdk"
-	"github.com/aws/aws-sdk-go-v2/internal/sync/singleflight"
 )
 
-// NeverExpire is the time identifier used when a credential provider's
-// credentials will not expire. This is used in cases where a non-expiring
-// provider type cannot be used.
-var NeverExpire = time.Unix(math.MaxInt64, 0)
-
-// AnonymousCredentials is an empty CredentialProvider that can be used as
-// dummy placeholder credentials for requests that do not need signed.
-//
-// This credentials can be used to configure a service to not sign requests
-// when making service API calls. For example, when accessing public
-// s3 buckets.
-//
-//     s3Cfg := cfg.Copy()
-//     s3cfg.Credentials = AnonymousCredentials
-//
-//     svc := s3.New(s3Cfg)
-var AnonymousCredentials = StaticCredentialsProvider{
-	Value: Credentials{Source: "AnonymousCredentials"},
-}
-
-// An Expiration provides wrapper around time with expiration related methods.
-type Expiration time.Time
-
-// Expired returns if the time has expired.
+//// AnonymousCredentials is an empty CredentialProvider that can be used as
+//// dummy placeholder credentials for requests that do not need signed.
+////
+//// This credentials can be used to configure a service to not sign requests
+//// when making service API calls. For example, when accessing public
+//// s3 buckets.
+////
+////     s3Cfg := cfg.Copy()
+////     s3cfg.Credentials = AnonymousCredentials
+////
+////     svc := s3.New(s3Cfg)
+//var AnonymousCredentials = StaticCredentialsProvider{
+//	Value: Credentials{Source: "AnonymousCredentials"},
+//}
```

Comment:
I know we've discussed this briefly, do we want to continue to have an Anonymous  sentinel value, or do we want make the statement that nil credentials == anonymous credentials?

Reply:
Thanks for pointing that out. I think API client's signer should treat nil credentials as anonymous.

This is slightly more complicated of a story for the `config.LoadDefaultAWSConfig` behavior, since `nil` `aws.Config.Credentials` is treated as unset, and can be populated.  This usability story probably shouldn't drive the behavior downstream at the API client level though, and we can solve separately.

Reply:
Discussed this further, and decided to keep the anonymous credentials. the signer middleware should not perform signing if credentials are `nil` or `Anonymous`. Having the `Anonymous` sentinel, is very helpful for configuring the SDK's external config loading.